backend/composer/admin.py

```diff
@@ -89,26 +97,32 @@ class SentenceAdmin(
         NoteSentenceInline,
     )
 
-
-class SynonymInline(admin.TabularInline):
-    model = Synonym
-    extra = 1
-
-
 class AnatomicalEntityAdmin(admin.ModelAdmin):
     search_fields = ('simple_entity__name', 'region_layer__layer__name', 'region_layer__region__name')
+    autocomplete_fields = ('simple_entity', 'region_layer')
+    list_display = ('simple_entity', 'region_layer', "synonyms")
+    list_display_links = ('simple_entity', 'region_layer')
+    inlines = (SynonymeInline,)
 
-    def get_model_perms(self, request):
-        """
-        Return empty dict to hide the model from admin index.
-        """
-        return {}
+    # we need to make efficient queries to the database to get the list of anatomical entities
+    def get_queryset(self, request: HttpRequest) -> QuerySet[Any]:
+        return super().get_queryset(request) \
+            .select_related('simple_entity', 'region_layer__layer', 'region_layer__region') \
+            .prefetch_related('synonyms')
+
+    @admin.display(description="Synonyms")
+    def synonyms(self, obj):
+        synonyms = obj.synonyms.all()
+        return ', '.join([synonym.name for synonym in synonyms])
 
 
 class AnatomicalEntityMetaAdmin(admin.ModelAdmin):
     list_display = ("name", "ontology_uri")
     list_display_links = ("name", "ontology_uri")
-    search_fields = ("name",)
+    search_fields = ("name", "ontology_uri")
+
+    def get_model_perms(self, request):
```

Comment:
question: @D-GopalKrishna why is this needed? can't we just remove the register (at the bottom)?

Reply:
No, django admin requires it if we need to use - `autocomplete_fields`

![image](https://github.com/MetaCell/sckan-composer/assets/59233227/793d6e49-c95f-4958-8e2b-9b93c64bfe3b)